### PR TITLE
Make glob work for windows

### PR DESCRIPTION
**Pull request by zackradisic**
WIP 2 tests are still broken don't merge yet

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
glob\scan and glob\leak

File: src/bun.js/api/glob.zig [hunk 1]
```diff
@@ -42,41 +42,41 @@ const ScanOpts = struct {
...
+            //             return null;
+            //         };
+
+            //         bun.strings.copyU8IntoU16(output, cwd_zig_str.slice());
+            //         break :brk output;
+            //     };
+
+            //     const ptr: [*]u8 = @ptrCast(utf16.ptr);
+            //     break :cwd_str_raw ZigString.Slice.from(ptr[0 .. utf16.len * 2], allocator);
+            // }
```
Comment (jdalton) [inline src/bun.js/api/glob.zig]: debug ☝️ ?

File: src/glob.zig [hunk 2]
```diff
@@ -1005,12 +1054,14 @@ pub fn GlobWalker_(
...
-            } else {
-                return filepath[1] == '.';
-            }
+        fn startsWithDot(filepath: []const u8) bool {
+            // DirIterator actually converts to UTF-8 internally so no need for this
+            // if (comptime !isWindows) {
+            //     return filepath.len > 0 and filepath[0] == '.';
+            // } else {
+            //     return filepath.len > 1 and filepath[1] == '.';
+            // }
```
Comment (jdalton) [inline src/glob.zig]: ☝️ Debug?

File: scripts/download-zls.ps1 [hunk 3]
```diff
@@ -2,6 +2,8 @@ push-location .cache
 try {
     git clone https://github.com/zigtools/zls
     set-location zls
-    git checkout 62f17abe283bfe0ff2710c380c620a5a6e413996
+    git pull
```
Comment (paperclover) [inline scripts/download-zls.ps1]: "git pull" followed by a checkout doesnt really make sense, but if this works that's fine. this whole setup needs an overhaul later

File: src/glob.zig [hunk 4]
```diff
@@ -109,8 +110,26 @@ const CursorState = struct {
...
 };
 
+const log = bun.Output.scoped(.glob, false);
+
 pub const BunGlobWalker = GlobWalker_(null, false);
 
+fn getcwd(buf: *[bun.MAX_PATH_BYTES]u8) Maybe([]const u8) {
+    if (bun.Environment.isWindows) {
+        var wbuf: bun.WPathBuffer = undefined;
+        const ptr = bun.windows._wgetcwd(&wbuf, wbuf.len) orelse {
```
Comment (paperclover) [inline src/glob.zig]: is using c-runtime required for this? can [GetCurrentDirectory](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getcurrentdirectory) or `bun.getcwd` be used instead?

actually i see that wgetcwd is using wide chars, while the windows api is ansi. hmm.

**Review by paperclover — COMMENTED** [review]
i guess i just have questions about the new glob.zig `getcwd` helper, but this looks alright otherwise.

File: scripts/download-zls.ps1 (hunk 3 above)
Comment (gvilums) [reply scripts/download-zls.ps1]: these changes shouldn't even be part of this PR. I'll remove them

File: src/glob.zig (hunk 4 above)
Comment (gvilums) [reply src/glob.zig]: switched to using fs.top_level_dir instead

**Review by gvilums — APPROVED** [review]

**Pull request merged**